include/kernel.h

```diff
@@ -3114,7 +3114,8 @@ extern int k_delayed_work_submit_to_queue(struct k_work_q *work_q,
  * @param work Address of delayed work item.
  *
  * @retval 0 Work item countdown canceled.
- * @retval -EINVAL Work item is being processed or has completed its work.
+ * @retval -EINVAL Work item is being processed.
```

Comment:
It will be EINVAL if it has never been submitted. I would regard this as the same case as already processed.
So maybe 
```
int k_delayed_work_cancel(struct k_delayed_work *work)
{
	if (!work->work_q) {
		return -EINVAL;
	}
```
Should return -EALREADY?

I think that is easier to document.
EINVAL: work item is being processed and cannot be canceled.
EALRADY: Work item is not pending

Reply:
Yep, it probably make sense to have them both returning -EALREADY so the application can safely resubmit if needs to, as oppose to -EINVAL which indicates that something went wrong and the work is still in progress.